### PR TITLE
Move all header buttons to overflow menu

### PR DESCRIPTION
**Pull request by srikarparsi**
### Details

Screenshots:

The join and leave thread buttons will be the [chatbubbles icon](https://expensify.slack.com/archives/C05RECHFBEW/p1695228649360879?thread_ts=1694794809.980809&cid=C05RECHFBEW) but there's placeholders for now.

image
image

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/318732
PROPOSAL: 

### Tests

1. Click pin and unpin conversation to make sure it works as expected
2. Click the google meets and zoom buttons to make sure they take you to their respective site
3. Click on a thread
4. Click join thread and ensure that the thread still shows up in most recent mode even when you navigate away.
5. Go back and click leave thread and ensure the thread disappears from your LHN in most recent mode when you navigate away.

- [ ] Verify that no errors appear in the JS console

### QA Steps

Same as tests.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/48188732/84d77db9-9c41-4848-a0c8-0731653f172a

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/48188732/61350bd0-f263-4ba7-baac-742618f792df

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/48188732/9486df89-7a14-41e0-906c-7bc20460f4c8

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/48188732/14f87c1e-2fa8-4e0f-9d9f-416b2bb9e247

</details>

<details>
<summary>iOS</summary>

IMG_8610
IMG_45A024F955FF-1

</details>

<details>
<summary>Android</summary>

Having trouble with emulator, will try more tonight

</details>

### CODE REVIEW

**Comment by srikarparsi** [comment]
Still need to translate the messages in spanish.

**Comment by shawnborton** [comment]
Screenshots please!

**Comment by srikarparsi** [comment]
Added :)

**Comment by shawnborton** [comment]
Nice, looks good so far - I like the leave icon too.

**Comment by madmax330** [comment]
@srikarparsi can you pull in a C+ 🙏🏽

File: src/libs/actions/Report.js
Comment (mollfpr) [inline src/libs/actions/Report.js:1231]: @srikarparsi The function is pretty similar to `updateNotificationPreferenceAndNavigate`, I think we can refactor it and reuse it between this function.

**Comment by dannymcclain** [comment]
Here's the "comment bubbles" icon (you can also find it [in Figma here](https://www.figma.com/file/qQIIaxVUdtggrh8eg8bI5B/Expensify-Design-System-2022?type=design&node-id=336%3A1694&mode=design&t=OH5U3CcErMBWrf8n-1))

Comment Bubbles

**Comment by srikarparsi** [comment]
Ah cool, thanks!

**Review by dannymcclain — DISMISSED** [review]
This is looking good! Going to approve without the updated chat bubbles icon but it would be awesome if we could add that in to this.

Also wanted to just point out that while "Leave thread" shows up in a thread's overflow menu, "Leave room" does not show up in a room's overflow menu, only in the details panel. I don't think that was necessarily supposed to be addressed in this issue, but it's an inconsistency that's probably worth thinking about.

**Comment by srikarparsi** [comment]
Cool yeah! We actually have an [issue](https://github.com/Expensify/Expensify/issues/318329) for that and I'll work on that one right after this one's merged :)

**Comment by srikarparsi** [comment]
This is ready for re-review @mollfpr, I'm waiting on feedback for the translations but other than that this is ready to be looked at again.

**Comment by mollfpr** [comment]
Thanks @srikarparsi! I'll review the PR again in the morning. Meanwhile, could you complete the checklist and the recordings?

**Comment by srikarparsi** [comment]
Bump @mollfpr, looking to get this merged today because there is another PR held on this.

**Comment by mollfpr** [comment]
@srikarparsi On it now!

**Comment by srikarparsi** [comment]
Cool! I can work back and forth on this with you to get it merged asap if that works for you :)

**Comment by mollfpr** [comment]
Looks good! Completing the records now.

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by mollfpr — APPROVED** [review]
LGTM 👍

**Review by Julesssss — DISMISSED** [review]

**Review by Julesssss — COMMENTED** [review]
@srikarparsi did you ever get this working on your emulators? I'll kick off an Adhoc build to help test the missing native iOS/Android platforms.

**Comment by srikarparsi** [comment]
That would be great thanks @Julesssss! Haven't had time today to figure out the emulator bugs yet

**Comment by Julesssss** [comment]
Is your main device an iPhone? If so I'll take the Android AdHoc build and you can take the iOS one. Let me know if you need your iOS device needs to be added to our dev group.

**Comment by srikarparsi** [comment]
Yeah it is, is there an SO for setting up AdHoc builds? Did a quick search but couldn't find anything

**Comment by Julesssss** [comment]
@srikarparsi you use the links above that I generated. Happy to explain how to do this in a  StackOverflow post if you write a question.

**Comment by srikarparsi** [comment]
Cool just asked: https://stackoverflowteams.com/c/expensify/questions/16018. 

I was trying to download the build using the QR code yesterday but have been getting this error:
IMG_03C2EA7CCAA7-1

**Review by Julesssss — COMMENTED** [review]
One issue I noticed is that when leaving a thread we'll navigate you to the concierge chat which seems wrong. But let me know if you think that existed prior to this PR: 

https://github.com/Expensify/App/assets/10736861/ca2a752e-7a00-48d6-83fc-8590e1ade3ff

**Comment by srikarparsi** [comment]
Yeah it did exist prior to this PR but it does seem to be a bug based on this code: https://github.com/Expensify/App/blob/main/src/libs/actions/Report.js#L1897

I'll make a post about it in expensify bugs

**Comment by Julesssss** [comment]
> I was trying to download the build using the QR code yesterday but have been getting this error: !

Answered [here](https://stackoverflowteams.com/c/expensify/questions/17469/17470#17470). The steps are a bit annoying but you only have to do it once!

**Comment by MitchExpensify** [comment]
@shawnborton Taggin' ya

**Comment by shawnborton** [comment]
It looks like we're missing the overflow button on expense reports. I would think that we'd at least have the pin + call options in that menu, since they are no different than any other room.

**Comment by shawnborton** [comment]
Same for expense views: 
image

**Comment by srikarparsi** [comment]
Hmm, I'm not sure the design that went into this but this is the behavior on main as well. (To not include pin and call on expense reports and views) I think if we decide to add the pin and call buttons to expense reports and expense views, it might be better handled in a follow up issue?

**Comment by shawnborton** [comment]
Ah okay, if that exists on main then let's create a follow up issue to treat all of the rooms the same. Does that sound right @JmillsExpensify @trjExpensify ?

**Comment by trjExpensify** [comment]
Yeah, I thought this initiative was to add those options to all reports and within the overflow menu tbh. A separate PR for it sounds good though.

**Comment by srikarparsi** [comment]
Cool created an [issue](https://github.com/Expensify/Expensify/issues/320550) and assigned myself to it. I think the main initiative is to be able to join and leave threads and consolidating the header overflow menu was a side effect of that since we needed to add the `join` option to it for threads.

**Comment by srikarparsi** [comment]
[Waiting to get access on Apple Developer](https://expensify.slack.com/archives/C03TQ48KC/p1695800020151949) to create a build on my phone and add screenshots. I think we can merge after that.

**Review by Julesssss — APPROVED** [review]
Looks good, we just need to test iOS. You should be able to continue with the physical iOS build stackoverflow, right? Else I can test on my test device later this evening

**Comment by srikarparsi** [comment]
Just added the iOS screenshots so this should be ready to merge whenever you have a chance :)

**Comment by srikarparsi** [comment]
Responded on the [issue](https://github.com/Expensify/App/issues/29851#issuecomment-1771831415)

File: src/pages/home/HeaderView.js
Comment (eVoloshchak) [inline src/pages/home/HeaderView.js:148]: 👋 Coming from https://github.com/Expensify/App/issues/28535
This doesn't need an `iconFill`, the color is already applied by default, but passing it here was overriding the behavior when the icon changes color when you hover over it

**Pull request merged**
